File: quickSort.js
function createRandomArray(num) {
	newArray = [];
	for (var i = 0; i < num; i++) {
		newArray[i] = Math.floor((Math.random()*10)+1);
	}
	return newArray;
}

function quickSort(origArray) {
	if (origArray.length <= 1) { 
		return origArray;
	} else {

		var left = [];
		var right = [];
		var newArray = [];
		var pivot = origArray.pop();
		var length = origArray.length;

		for (var i = 0; i < length; i++) {
			if (origArray[i] <= pivot) {
				left.push(origArray[i]);
			} else {
				right.push(origArray[i]);
			}
		}

		return newArray.concat(quickSort(left), pivot, quickSort(right));
	}
}

var myArray = createRandomArray(5);
console.log("Unsorted array: " + myArray);
var sortedArray = quickSort(myArray);
console.log("Sorted array: " + sortedArray);

File: quicksort.rb
def createRandomArray(num)
	newArray = []
	i = 0
	while i < num
		newArray[i] = (1 + rand(10))
		i += 1
	end
	newArray
end

def quickSort(origArray)
	if origArray.count <= 1
		return origArray
	else
		left = []
		right = []
		newArray = []
		pivot = origArray.pop
		length = origArray.count

		origArray.each do |element|
			if element <= pivot
				left << element
			else
				right << element
			end
		end

		newArray << quickSort(left) << pivot << quickSort(right)

		return newArray.flatten!
	end
end

myArray = createRandomArray(5)
puts "Unsorted array: #{myArray}"
sortedArray = quickSort(myArray)
puts "Sorted array: #{sortedArray}"